Api/Src/Controllers/TestController.ts

```diff
@@ -0,0 +1,41 @@
+import { Request, Response } from "express";
+import * as mongoose from 'mongoose';
+import TestService from '../Services/TestService';
+import { Test } from '../Models/Section';
+import SectionService from "../Services/SectionService";
+
+export default class TestController {
+    testService: TestService;
+    sectionService: SectionService;
+    constructor(testService: TestService, sectionService: SectionService) {
+        this.testService = testService;
+        this.sectionService = sectionService;
+    }
+
+    addTest = async (req: Request, res: Response) => {
+        const sectionId = new mongoose.Types.ObjectId(req.params.id);
+        const test = req.body;
+        const section = await this.testService.addTest(sectionId, test);
+        if(!section) return res.status(404).json({message: 'Section not found'});
+        res.status(200).json({message: 'Test added to section'});
+    }
+
+    updateTest = async (req: Request, res: Response) => {
+        const sectionId = new mongoose.Types.ObjectId(req.params.sectionId);
+        const testId = new mongoose.Types.ObjectId(req.params.testId);
+        const test = req.body;
+        const section = await this.sectionService.getSectionById(sectionId);
+        if (!section) res.status(404).json({message: 'Section not found'});
+        await this.testService.updateTest(sectionId, testId, test);
+        res.status(200).json({message: 'Test was updated'});
+    }
+
+    deleteTest = async (req: Request, res: Response) => {
+        const sectionId = new mongoose.Types.ObjectId(req.params.sectionId);
+        const testId = new mongoose.Types.ObjectId(req.params.testId);
+        const section = await this.sectionService.getSectionById(sectionId);
+        if (!section) res.status(404).json({message: 'Section not found'});
+        await this.testService.deleteTest(sectionId, testId);
```

Comment:
some condition to check if test exists would be good here - at the moment removing non-existing test ends with 200

Reply:
Right, I added new condition to check if test exists.